src/suricata.c

```diff
@@ -2671,6 +2682,11 @@ int PostConfLoadedSetup(SCInstance *suri)
     RunModeEngineIsIPS(
             suricata.run_mode, suricata.runmode_custom_mode, suricata.capture_plugin_name);
 
+    if (EngineModeIsUnknown()) { // if still uninitialized the set the default
```

Comment:
Is this really a bug?

Reply:
No this is meant to ensure the default engine mode is set when it hasn't been set until now.

But the whole PR is about making sure that Suricata modules don't query engine mode while uninitialized. 
If it is not initialized by other Suricata ways (e.g. config, command-line parameter) then set the default. But after this line, the engine mode (should) never change.